scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc

```diff
@@ -1711,3 +1713,436 @@ JNIEXPORT jint JNICALL Java_ml_dmlc_mxnet_LibInfo_mxRtcFree
+                                                       *(const_cast<bool*>(&isTrain)));
+                if ((int)result == 0) success = false;
+                globalEnv->DeleteLocalRef(tagsArr);
+                globalEnv->DeleteLocalRef(reqsArr);
+                globalEnv->DeleteLocalRef(ptrsArr);
+              }
+              _jvm->DetachCurrentThread();
+            }
+            return success;
+          };
+
+          // backward callback
+          auto backwardEntry = [](int size, void **ptrs, int *tags,
+            const int *reqs, const bool isTrain, void *state) {
+            std::string key((char *)state);
+            bool success = true;
+            // std::cout << "bkk" << std::endl;
+            if (globalOpMap.find(key) == globalOpMap.end()) {
+              std::cout << "op: " << key << " not found" << std::endl;
+              success = false;
+            } else {
+              _jvm->AttachCurrentThread((void **)&globalEnv, NULL);
+              jclass opClass = globalEnv->GetObjectClass(globalOpMap.at(key));
+              jmethodID midBackward = globalEnv->GetMethodID(opClass, "backwardEntry", "(I[J[I[IZ)I");
+              if (NULL == midBackward) {
+                std::cout << "could not find op method backwardEntry.";
+                success = false;
+              } else {
+                jintArray tagsArr = globalEnv->NewIntArray(size);
+                globalEnv->SetIntArrayRegion(tagsArr, (jsize)0, (jsize)size, tags);
+
+                int reqSize = 0;
+                for (int i=0; i<size; ++i) {
+                  if (tags[i] == 2) reqSize ++;
+                }
+                jintArray reqsArr = globalEnv->NewIntArray(reqSize);
+                globalEnv->SetIntArrayRegion(reqsArr, (jsize)0, (jsize)reqSize, reqs);
+                jlongArray ptrsArr = globalEnv->NewLongArray(size);
+                globalEnv->SetLongArrayRegion(ptrsArr, (jsize)0, (jsize)size, reinterpret_cast<jlong*>(ptrs));
+                jint result = globalEnv->CallIntMethod(globalOpMap.at(key), midBackward,
+                                                       size,
+                                                       ptrsArr,
+                                                       tagsArr,
+                                                       reqsArr,
+                                                       *(const_cast<bool*>(&isTrain)));
+                if ((int)result == 0) success = false;
+                globalEnv->DeleteLocalRef(tagsArr);
+                globalEnv->DeleteLocalRef(reqsArr);
+                globalEnv->DeleteLocalRef(ptrsArr);
+              }
+              _jvm->DetachCurrentThread();
+            }
+            return success;
+          };
+
+          // del callback
+          auto delEntry = [](void *state) {
+            std::string key((char *)state);
+            bool success = true;
+            // if (globalOpMap.find(key) == globalOpMap.end()) {
```

Comment:
actually I use global variable map to store the customop and customopprop object, and want to
remove them from the map and release them in the del fuction. But this delEntry function seems to be called during the execution, and cause the , so I comment the code.